testing/integration/upgrade_test.go

```diff
@@ -899,3 +899,89 @@ func removePackageVersionFiles(t *testing.T, f *atesting.Fixture) {
 		require.NoErrorf(t, err, "error removing package version file %q", vFile)
 	}
 }
+
+// TestStandaloneUpgradeFailsStatus tests the scenario where upgrading to a new version
+// of Agent fails due to the new Agent binary reporting an unhealthy status. It checks
+// that the Agent is rolled back to the previous version.
+func TestStandaloneUpgradeFailsStatus(t *testing.T) {
+	define.Require(t, define.Requirements{
+		Local:   false, // requires Agent installation
+		Isolate: false,
+		Sudo:    true, // requires Agent installation
+	})
+
+	upgradeFromVersion, err := version.ParseVersion(define.Version())
+	require.NoError(t, err)
+
+	// We go back TWO minors because sometimes we are in a situation where
+	// the current version has been advanced to the next release (e.g. 8.10.0)
+	// but the version before that (e.g. 8.9.0) hasn't been released yet.
+	previousVersion, err := upgradeFromVersion.GetPreviousMinor()
+	require.NoError(t, err)
+	previousVersion, err = previousVersion.GetPreviousMinor()
+	require.NoError(t, err)
```

Comment:
I would really prefer if we didn't try to guess what the older versions of agents are and instead query the artifact api for versions....

Reply:
Totally agree! Fixed in a98f49588c2073bdc06a99960b3c67f9a5a1d253.